### PR TITLE
BTAT-5950 Added hidden input fields for additional form fields and added to session

### PR DESCRIPTION
**Pull request by TomDudley92**

### CODE REVIEW

File: app/forms/SubmitVatReturnForm.scala [hunk 1]
```diff
@@ -32,8 +32,12 @@ object NineBoxForm {
       "box6" -> bigDecimal,
       "box7" -> bigDecimal,
       "box8" -> bigDecimal,
-      "box9" -> bigDecimal
-    )(NineBoxModel.apply)(NineBoxModel.unapply)
+      "box9" -> bigDecimal,
+      "flatRateScheme" -> boolean,
```
Comment (miranda-hawkes) [inline app/forms/SubmitVatReturnForm.scala]: These values don't necessarily need to be in the form and as hidden fields on the view - they could just be added to session separate to the 9 box values

File: test/views/ConfirmSubmissionViewSpec.scala [hunk 2]
```diff
@@ -99,7 +97,7 @@ class ConfirmSubmissionViewSpec extends ViewBaseSpec {
           }
 
           s"have the redirect url as ${controllers.routes.HelloWorldController.helloWorld()}" in {
```
Comment (miranda-hawkes) [inline test/views/ConfirmSubmissionViewSpec.scala]: Please rename test to correct url

File: test/views/ConfirmSubmissionViewSpec.scala [hunk 3]
```diff
@@ -179,7 +177,8 @@ class ConfirmSubmissionViewSpec extends ViewBaseSpec {
         "have the change return details link which" should {
 
           s"the redirect url to URL NEEDED" in {
```
Comment (miranda-hawkes) [inline test/views/ConfirmSubmissionViewSpec.scala]: Please rename test to correct url

File: app/forms/SubmitVatReturnForm.scala (hunk 1 above)
Comment (miranda-hawkes) [reply app/forms/SubmitVatReturnForm.scala]: Agreed to reduce complexity of code to retain in form

**Comment by miranda-hawkes** [comment]
Please fix merge conflicts

File: app/controllers/ConfirmSubmissionController.scala [hunk 4]
```diff
@@ -16,33 +16,49 @@
...
+        }
+      }
+      case _ => Future.successful(Redirect(controllers.routes.SubmitFormController.show(periodKey)))
+    }
   }
 
-  def submit(periodKey: String): Action[AnyContent] = Action.async { implicit request =>
-    Future.successful(Redirect(controllers.routes.ConfirmationController.show().url))
+  def submit(periodKey: String): Action[AnyContent] = (authPredicate andThen mandationStatusCheck) { implicit user =>
+    Redirect(controllers.routes.ConfirmationController.show().url)
```
Comment (miranda-hawkes) [inline app/controllers/ConfirmSubmissionController.scala]: Could you add removal of session data in this please? Then when we do the actual backend submission we don't forget to clear it after it being successful

Comment (TomDudley92) [reply app/controllers/ConfirmSubmissionController.scala]: This is being done under another task

File: app/controllers/SubmitFormController.scala [hunk 5]
```diff
@@ -38,45 +48,101 @@ class SubmitFormController @Inject()(val messagesApi: MessagesApi,
...
-      NineBoxForm.nineBoxForm.bindFromRequest().fold(
-        failure =>
-          Future.successful(
-            Ok(
-              views.html.submit_form(periodKey, name, hasFlatRateScheme, Json.parse(obligation).as[VatObligations], failure, user.isAgent)
-            )
-          ),
-        success => Future.successful(
-          Redirect(controllers.routes.ConfirmSubmissionController.show(hasFlatRateScheme, obligation, Json.toJson(success).toString, name, periodKey))
+  def submit(periodKey: String,
```
Comment (miranda-hawkes) [inline app/controllers/SubmitFormController.scala]: Please remove query parameters and retrieve again if page is invalid as discussed

File: app/views/submit_form.scala.html [hunk 6]
```diff
@@ -21,21 +21,37 @@
...
 @(periodKey: String,
-  clientName: Option[String],
-  flatRateScheme: Boolean,
-  obligations: VatObligations,
-  form: Form[NineBoxModel],
-  isAgent: Boolean)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+        clientName: Option[String],
+        flatRateScheme: Boolean,
+        obligation: VatObligation,
+        form: Form[SubmitVatReturnModel],
```
Comment (miranda-hawkes) [inline app/views/submit_form.scala.html]: The FRS and obligation data is now passed in twice to the view (as separate parameters and within the form model). Can they not be removed from the form entirely?

**Pull request merged**